unit-tests/src/main/scala/java/util/regex/RegexPatternSuite.scala

```diff
@@ -0,0 +1,16 @@
+package java.util.regex
+
+object RegexPatternSuite extends tests.Suite {
+  test("Expects abc to match abc") {
+    assert(Pattern.matches("abc", "abc"))
+  }
+  test("Expects [[:digit:]] to match 123") {
+    assert(Pattern.matches("[[:digit:]]", "123"))
+  }
+  test("Expects [[:digit:]] to match 1") {
+    assert(Pattern.matches("[[:digit:]]", "1")) // This is segfaulting!
+  }
+  test("Expects Scal.-Native to match Scala-Native") {
+    assert(Pattern.matches("Scal.-Native", "Scala-Native"))
+  }
+}
```

Comment:
More tests would be really welcome here. Perhaps have a look at https://github.com/scala-js/scala-js/blob/master/test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/regex/RegexPatternTest.scala